packages/context/test/unit/context.unit.ts

```diff
@@ -681,6 +685,109 @@ describe('Context', () => {
     });
   });
 
+  describe('listener subscription', () => {
+    let nonMatchingListener: ContextEventListener;
+
+    beforeEach(givenNonMatchingListener);
+
+    it('subscribes listeners', () => {
+      ctx.subscribe(nonMatchingListener);
+      expect(ctx.isSubscribed(nonMatchingListener)).to.true();
+    });
+
+    it('unsubscribes listeners', () => {
+      ctx.subscribe(nonMatchingListener);
+      expect(ctx.isSubscribed(nonMatchingListener)).to.true();
+      ctx.unsubscribe(nonMatchingListener);
+      expect(ctx.isSubscribed(nonMatchingListener)).to.false();
+    });
+
+    it('allows subscription.unsubscribe()', () => {
+      const subscription = ctx.subscribe(nonMatchingListener);
+      expect(ctx.isSubscribed(nonMatchingListener)).to.true();
+      subscription.unsubscribe();
+      expect(ctx.isSubscribed(nonMatchingListener)).to.false();
+      expect(subscription.closed).to.be.true();
+    });
+
+    it('registers listeners on context chain', () => {
+      const childCtx = new Context(ctx, 'child');
+      childCtx.subscribe(nonMatchingListener);
+      expect(childCtx.isSubscribed(nonMatchingListener)).to.true();
+      expect(ctx.isSubscribed(nonMatchingListener)).to.true();
+    });
+
+    it('un-registers listeners on context chain', () => {
+      const childCtx = new Context(ctx, 'child');
+      childCtx.subscribe(nonMatchingListener);
+      expect(childCtx.isSubscribed(nonMatchingListener)).to.true();
+      expect(ctx.isSubscribed(nonMatchingListener)).to.true();
+      childCtx.unsubscribe(nonMatchingListener);
+      expect(childCtx.isSubscribed(nonMatchingListener)).to.false();
+      expect(ctx.isSubscribed(nonMatchingListener)).to.false();
+    });
+
+    function givenNonMatchingListener() {
+      nonMatchingListener = {
+        filter: binding => false,
+        listen: (event, binding) => {},
+      };
+    }
+  });
+
+  describe('event notification', () => {
+    let matchingListener: ContextEventListener;
+    let nonMatchingListener: ContextEventListener;
+    const events: string[] = [];
+    let nonMatchingListenerCalled = false;
+
+    beforeEach(givenListeners);
+
+    it('emits bind event to matching listeners', async () => {
+      ctx.bind('foo').to('foo');
+      await nextTick();
```

Comment:
I find this rather brittle, it's relying on the internal implementation details of Context class. Can we find a more robust solution please? See my first comment above for a possible option.

Reply:
Fixed